provider/pkg/provider/yaml/v2/yaml.go

```diff
@@ -140,14 +182,54 @@ type RegisterOptions struct {
 // Register registers the given Kubernetes objects as resources with the Pulumi engine.
 // Returns an array of the resources that were registered.
 func Register(ctx *pulumi.Context, opts RegisterOptions) (pulumi.ArrayOutput, error) {
-	resources := pulumi.Array{}
+	objs := make(cliutilsobject.UnstructuredSet, 0, len(opts.Objects))
 	for _, obj := range opts.Objects {
-		rs, err := register(ctx, &obj, opts)
-		if err != nil {
-			return pulumi.ArrayOutput{}, err
-		}
-		for _, r := range rs {
+		obj := obj
```

Comment:
If this is to avoid a loop capture bug, it's not needed unless there's a closure/goroutine involved. I'd love to set `GOEXPERIMENT=loopvar` and never think about these again.

Reply:
Since we store a pointer to the loop variable (into `objs`), I thought it necessary.

Reply:
We still need to do this since we're still using Go v1.21.
Fyi, the `GOEXPERIMENT=loopvar` preview is now the default in Go v1.22 (https://go.dev/blog/loopvar-preview). So we don't need to worry about these loop capture bugs once we upgrade to Go v1.22.